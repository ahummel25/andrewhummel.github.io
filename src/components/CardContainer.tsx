import React, { FC } from 'react';
import styled from '@emotion/styled';

import { fonts } from '../styles/variables';

type ContainerProps = {
  width?: number;
};

const Container = styled.div<ContainerProps>`
  width: 95%;
  font-family: ${fonts.roboto};
  font-weight: 300;
  padding: 0 24px;
  max-width: 1280px;

  @media (min-width: 1200px) {
    width: ${(props): number | undefined => props.width}px;
  }
`;

const CardContainer: FC<ContainerProps> = ({
  children,
  width
}): JSX.Element => (
  <>
    <Container className="container" width={width}>
      {children}
    </Container>
  </>
);

export default CardContainer;
