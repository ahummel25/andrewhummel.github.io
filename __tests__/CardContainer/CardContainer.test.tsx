import React from 'react';
import { create } from 'react-test-renderer';

import CardContainer from '../../src/components/CardContainer';

describe('CardContainer', () => {
  it('renders correctly', () => {
    const tree = create(<CardContainer />);

    expect(tree).toBeDefined();
  });

  it('renders correctly with width prop', () => {
    const tree = create(<CardContainer width={860} />);
    const root = tree.root;

    expect(root.props).toStrictEqual({ width: 860 });
  });
});
