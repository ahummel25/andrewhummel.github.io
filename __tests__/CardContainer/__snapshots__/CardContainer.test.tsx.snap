// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer renders correctly 1`] = `
.emotion-0 {
  width: 95%;
  font-family: Roboto,sans-serif;
  font-weight: 300;
  padding: 0 24px;
  max-width: 1280px;
  letter-spacing: 0.25px;
}

@media (min-width: 1200px) {
  .emotion-0 {
    width: px;
  }
}

.emotion-0 {
  width: 95%;
  font-family: Roboto,sans-serif;
  font-weight: 300;
  padding: 0 24px;
  max-width: 1280px;
  letter-spacing: 0.25px;
}

@media (min-width: 1200px) {
  .emotion-0 {
    width: px;
  }
}

<div
  className="container emotion-0"
/>
`;
